library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -604,7 +647,12 @@ protected void replay(long windowId)
                 pendingFiles.remove(recoveryEntry.file);
               }
               inputStream = retryFailedFile(new FailedFile(recoveryEntry.file, recoveryEntry.startOffset));
-
+              if (skipCount == 0) {
```

Comment:
If the file processing failed just after opening the file without reading any records, we emit startFileControlTuple. Since we are reading from beginning, it indicates file processing had not failed earlier. This information is conveyed via the parameter (false) passed.